plugins/modules/lxd_container.py

```diff
@@ -468,11 +468,7 @@ def __init__(self, module):
         self.type = self.module.params['type']
 
         # LXD Rest API provides additional endpoints for creating containers and virtual-machines.
-        self.api_endpoint = None
-        if self.type == 'container':
-            self.api_endpoint = '/1.0/containers'
-        elif self.type == 'virtual-machine':
-            self.api_endpoint = '/1.0/virtual-machines'
+        self.api_endpoint = '/1.0/instances'
```

Comment:
This is likely a breaking change for older LXD versions.

Reply:
LXD has supported this endpoint since v4.0 (LTS) released April 3, 2020: https://github.com/canonical/lxd/blob/lxd-4.0.0/doc/api-extensions.md ([announcement](https://ubuntu.com/blog/lxd-4-0-lts-stable-release-is-now-available))

I'm not sure what backwards compatibility guarantees `community.general` makes or how best to preserve backwards compatibility with older versions. Can you provide guidance?

Reply:
LXD v3.0 LTS branch reached EOL in July 2023: https://discuss.linuxcontainers.org/t/lxc-lxd-lxcfs-3-0-end-of-life-announcement/17600

Reply:
The module unfortunately does not specify an explicit lower bound of the LXD versions it supports.

The best way would be to have some version detection and then a version based behavior selection (grep for `LooseVersion` in this repo to see how version comparisons can be done). I would guess there is an API endpoint which allows you to get info on LXD itself, including its version, that can be used for this.

Reply:
> The module unfortunately does not specify an explicit lower bound of the LXD versions it supports.
> 
> The best way would be to have some version detection and then a version based behavior selection (grep for `LooseVersion` in this repo to see how version comparisons can be done). I would guess there is an API endpoint which allows you to get info on LXD itself, including its version, that can be used for this.

# Instances, containers and virtual-machines

The documentation shows paths such as /1.0/instances/..., which were introduced with LXD 3.19. Older releases that supported only containers and not virtual machines supply the exact same API at /1.0/containers/....

For backward compatibility reasons, LXD does still expose and support that /1.0/containers API, though for the sake of brevity, we decided not to double-document everything.

An additional endpoint at /1.0/virtual-machines is also present and much like /1.0/containers will only show you instances of that type.

Reference: https://documentation.ubuntu.com/lxd/en/latest/rest-api/#instances-containers-and-virtual-machines

It appears that this plugin doesn't support the `/1.0/containers` so at which point should there just be a note that it won't work after X.X version?